drivers/wifi/nrf700x/zephyr/src/qspi/src/ficr_prog.c

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* @file
+ * @brief NRF Wi-Fi radio FICR programming functions
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr.h>
+#include "rpu_hw_if.h"
+#include "ficr_prog.h"
+
+
+LOG_MODULE_DECLARE(otp_prog, CONFIG_WIFI_LOG_LEVEL);
+
+static void write_word(unsigned int addr, unsigned int data)
+{
+	rpu_write(addr, &data, 4);
+}
+
+static void read_word(unsigned int addr, unsigned int *data)
+{
+	rpu_read(addr, data, 4);
+}
+
+unsigned int check_protection(unsigned int *buff, unsigned int off1, unsigned int off2,
+		    unsigned int off3, unsigned int off4)
+{
+	if ((buff[off1] == 0x00000000) &&
+		(buff[off2] == 0x00000000)  &&
+		(buff[off3] == 0x00000000) &&
+		(buff[off4] == 0x00000000))
+		return 0x00000000;
+	else if ((buff[off1] == 0xFFFFFFFF) &&
+		(buff[off2] == 0xFFFFFFFF) &&
+		(buff[off3] == 0xFFFFFFFF) &&
+		(buff[off4] == 0xFFFFFFFF))
+		return 0xFFFFFFFF;
+	else if ((buff[off1] == 0x50FA50FA) &&
+		(buff[off2] == 0x50FA50FA) &&
+		(buff[off3] == 0x50FA50FA) &&
+		(buff[off4] == 0x50FA50FA))
+		return 0x50FA50FA;
+	else
+		return 0xdeadbeef;
+}
+
+
+static void set_otp_timing_reg_40mhz(void)
+{
+	write_word(OTP_TIMING_REG1_ADDR, OTP_TIMING_REG1_VAL);
+
+	write_word(OTP_TIMING_REG2_ADDR, OTP_TIMING_REG2_VAL);
+}
+
+static int poll_otp_ready(void)
+{
+
+	int otp_mem_status = 0;
+	int poll = 0;
+
+	while (poll != 100) {
+		read_word(OTP_POLL_ADDR, &otp_mem_status);
+
+		if ((otp_mem_status&0x4) == OTP_READY) {
```

Comment:
```suggestion
		if ((otp_mem_status & OTP_READY) == OTP_READY) {
```
or better introduce #define OTP_READY_MASK 0x4